### PR TITLE
Allow passing query between egress filter and filter gateway.

### PR DESCRIPTION
**Pull request by mtanig**
### Description:

- Enable to pass query between egress filter and filter gateway.
- Add filter function examples.

### Versions:

- Go Version: 1.20.3
- Docker Version: 23.0.5
- Kubernetes Version: 1.27.1
- NGT Version: 2.0.9

### Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/main/CONTRIBUTING.md) document and completed [our CLA agreement](https://cla-assistant.io/vdaas/vald).
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?

### Special notes for your reviewer:

You can check the behavior after this modification by reading the example/client/gateway/filter/README.md .

In this case, you need to additionally build deploy of filter-gateway as follows.
```
vim docker.mk

docker/build/gateway-filter

-t $(ORG)/$(FILTER_GATEWAY_IMAGE):$(TAG) . \ -> -t mtani/$(FILTER_GATEWAY_IMAGE):$(TAG) . \


make docker/build/gateway-filter

docker login
docker push mtani/vald-filter-gateway:latest


vim example/helm/values.yaml

gateway:
  filter:
    enabled: true
    image:
      repository: mtani/vald-filter-gateway
      tag: latest
```

### CODE REVIEW

**Comment by hlts2** [comment]
@mtanig 
Thank you for PR!! What is the status of this PR? 
And could you please fix the conflict code? 🙏

**Comment by mtanig** [comment]
@hlts2 
There was a difference in the generated files in my environment, so please let me discuss it on Slack 🙇

File: pkg/gateway/filter/handler/grpc/handler.go [hunk 1]
```diff
@@ -1434,47 +1433,47 @@ func (s *server) Search(ctx context.Context, req *payload.Search_Request) (res *
...
-					RequestId:   req.GetConfig().GetRequestId(),
-					ServingData: errdetails.Serialize(req),
-				},
-				&errdetails.BadRequest{
-					FieldViolations: []*errdetails.BadRequestFieldViolation{
-						{
-							Field:       "vectorizer targets",
-							Description: err.Error(),
+	filterConfigs = req.GetConfig().GetEgressFilters()
+	if filterConfigs != nil || s.DistanceFilters != nil {
```
Comment (hlts2) [inline pkg/gateway/filter/handler/grpc/handler.go:1437]: Is it OK that this variable(`s.DistanceFilters`) does not seem to be used?

cc: @kevindiu

Comment (kevindiu) [reply pkg/gateway/filter/handler/grpc/handler.go:1437]: Maybe we can remove `s.DistanceFilters` as we are not using it anymore.

Comment (mtanig) [reply pkg/gateway/filter/handler/grpc/handler.go:1437]: As you pointed out, that description seems to have disappeared. 
I'm sorry, but would you mind correcting that? @kevindiu

**Pull request merged**